src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataSerializerContext.cs

```diff
@@ -312,7 +348,16 @@ internal IEdmTypeReference GetEdmType(object instance, Type type)
                 {
                     if (instance != null)
                     {
-                        edmType = _typeMappingCache.GetEdmType(instance.GetType(), Model);
+                        TypedDelta delta = instance as TypedDelta;
+
+                        if (delta != null)
```

Comment:
```suggestion
                        if (instance is TypedDelta delta)
```

Reply:
is not necesary to do the null check here.

Reply:
`instance is TypedDelta delta` returns false if `instance` is null